### PR TITLE
Aesthetic changes to heatmap and clone genome distribution plot

### PR DESCRIPTION
**Pull request by whelena**
# Description
Update aesthetic choices such as legend placement, adding a covariate heatmap option, legend ordering, etc. Make default plots prettier while allowing for customizations by adding parameters.

# Checklist

- [x] This PR **does *NOT* contain** Protected Health Information [(PHI)](https://ohrpp.research.ucla.edu/hipaa/). A repo may ***need to be deleted*** if such data is uploaded. <br> Disclosing PHI is a ***major problem***[^1] - Even ***a small leak can be costly***[^2].
  
- [ ] This PR **does *NOT* contain** germline genetic data[^3], RNA-Seq, DNA methylation, microbiome or other molecular data[^4].

[^1]: [UCLA Health reaches $7.5m settlement over 2015 breach of 4.5m patient records](https://healthitsecurity.com/news/ucla-health-reaches-7.5m-settlement-over-2015-breach-of-4.5m)
[^2]: [The average healthcare data breach costs $2.2 million, despite the majority of breaches releasing fewer than 500 records.](https://www.ponemon.org/local/upload/file/Sixth%20Annual%20Patient%20Privacy%20%26%20Data%20Security%20Report%20FINAL%206.pdf)
[^3]: [Genetic information is considered PHI.](https://www.genome.gov/about-genomics/policy-issues/Privacy#:~:text=In%202013%2C%20as%20required%20by,genetic%20information%20for%20underwriting%20purposes.)
  [Forensic assays can identify patients with as few as 21 SNPs](https://www.sciencedirect.com/science/article/pii/S1525157817305962)
[^4]: [RNA-Seq](https://www.nature.com/articles/ng.2248), [DNA methylation](https://ieeexplore.ieee.org/document/7958619), [microbiome](https://www.pnas.org/doi/pdf/10.1073/pnas.1423854112), or other molecular data can be used to predict genotypes (PHI) and reveal a patient's identity.

- [x] This PR **does *NOT* contain** other non-plain text files, such as: compressed files, images (*e.g.* `.png`, .`jpeg`), `.pdf`, `.RData`, `.xlsx`, `.doc`, `.ppt`, or other output files.

_&emsp; To automatically exclude such files using a [.gitignore](https://docs.github.com/en/get-started/getting-started-with-git/ignoring-files) file, see [here](https://github.com/uclahs-cds/template-base/blob/main/.gitignore) for example._

- [ ] I have read the [code review guidelines](https://uclahs-cds.atlassian.net/wiki/spaces/BOUTROSLAB/pages/3187646/Code+Review+Guidelines) and the [code review best practice on GitHub check-list](https://uclahs-cds.atlassian.net/wiki/spaces/BOUTROSLAB/pages/3189956/Code+Review+Best+Practice+on+GitHub+-+Check+List).

- [x] I have set up or verified the `main` branch protection rule following the [github standards](https://uclahs-cds.atlassian.net/wiki/spaces/BOUTROSLAB/pages/3190380/GitHub+Standards#GitHubStandards-Branchprotectionrule) before opening this pull request.

- [x] The name of the branch is meaningful and well formatted following the [standards](https://uclahs-cds.atlassian.net/wiki/spaces/BOUTROSLAB/pages/3189956/Code+Review+Best+Practice+on+GitHub+-+Check+List), using [AD_username (or 5 letters of AD if AD is too long)]-[brief_description_of_branch].
  
- [x] I have added the major changes included in this pull request to the `CHANGELOG.md` under the next release version or unreleased, and updated the date.

### CODE REVIEW

File: R/create.ccf.summary.heatmap.R [hunk 1]
```diff
@@ -49,30 +65,38 @@ create.ccf.summary.heatmap <- function(
...
     sample.bar <- BoutrosLab.plotting.general::create.barplot(
         formula = ID ~ nsnv,
         data = sample.df,
-        xlab.label = 'SNV per sample',
+        xlab.label = 'SNV\nper sample',
         xlab.cex = subplot.xlab.cex,
         xaxis.cex = subplot.xaxis.cex,
         xaxis.fontface = subplot.xaxis.fontface,
-        xlimits = c( - max(sample.df$nsnv) * 0.05, max(sample.df$nsnv) * 1.05),
+        xlimits = c( - 0.05, 1.05) * max(sample.xaxis$at),
```
Comment (dan-knight) [inline R/create.ccf.summary.heatmap.R:85]: Nice readability improvement.

File: R/create.ccf.summary.heatmap.R [hunk 2]
```diff
@@ -88,27 +112,56 @@ create.ccf.summary.heatmap <- function(
...
+            input.colours = TRUE,
+            clustering.method = 'none',
+            grid.col = FALSE,
+            print.colour.key = FALSE,
+            yaxis.tck = 0
+            );
+        plot.list <- list(clone.bar, hm, sample.bar, clone.cov);
+        layout.skip <- c(FALSE, TRUE, FALSE, FALSE, FALSE, TRUE);
+        layout.height <- 3;
+        plot.objects.heights <- c(0.3, 0.9, 0.2);
```
Comment (dan-knight) [inline R/create.ccf.summary.heatmap.R:145]: Can we avoid the magic numbers for `plot.objects.heights`? Just some named variables might help for clarity.

File: R/create.ccf.summary.heatmap.R [hunk 3]
```diff
@@ -45,7 +45,7 @@ create.ccf.summary.heatmap <- function(
     names(sample.df)[2] <- names(clone.df)[2] <- 'nsnv';
 
     if (!is.null(clone.order) & !is.null(sample.order)) {
-        arr                 <- arr[clone.order, sample.order];
+        arr                 <- arr[clone.order, rev(sample.order)];
```
Comment (dan-knight) [inline R/create.ccf.summary.heatmap.R:49]: I assume this is because the heatmap goes bottom-to-top>

Comment (whelena) [reply R/create.ccf.summary.heatmap.R:49]: yepp

**Review by dan-knight — APPROVED** [review]

**Pull request merged**